### PR TITLE
docs: fix book hyperlink for depends-on

### PR DESCRIPTION
**Pull request by sdowell**
The hyperlink was missing a trailing / which was leading to a 404 for
the redirected page.

### CODE REVIEW

File: site/book/06-deploying-packages/03-handling-dependencies.md [hunk 1]
```diff
@@ -83,4 +83,4 @@ service/wordpress-mysql deleted
 See [depends-on] for more information.
 
 [depends-on]:
-  /reference/annotations/depends-on
+  /reference/annotations/depends-on/
```
Comment (phanimarupaka) [inline site/book/06-deploying-packages/03-handling-dependencies.md:86]: This should be fixed in general and not just for this link. Why is this happening only for this and not for other links. e.g. https://raw.githubusercontent.com/GoogleContainerTools/kpt/main/site/installation/migration.md All links doesn't have trailing slash

Comment (sdowell) [reply site/book/06-deploying-packages/03-handling-dependencies.md:86]: I don't think there is anything to "fix" per se - rather it is a nuance of the tool we are using to render the web site from the markdown files. It seems a trailing slash indicates a directory path at which the default markdown file (e.g. `README.md`) located in that directory will be rendered on the web page. A few examples:

ex 1:
markdown: https://github.com/GoogleContainerTools/kpt/blob/main/site/book/README.md
URL: https://kpt.dev/book/

ex 2:
markdown: https://github.com/GoogleContainerTools/kpt/blob/main/site/installation/README.md
URL: https://kpt.dev/installation/

Whereas when referencing an explicit markdown file rather than a directory, the trailing slash is omitted. A few examples:

ex 1:
markdown: https://github.com/GoogleContainerTools/kpt/blob/main/site/installation/migration.md
URL: https://kpt.dev/installation/migration

ex 2:
markdown: https://github.com/GoogleContainerTools/kpt/blob/main/site/book/06-deploying-packages/03-handling-dependencies.md
URL: https://kpt.dev/book/06-deploying-packages/03-handling-dependencies

TL;DR it seems it was a mistake to omit the trailing slash in this case since the intent is to reference the `README.md` file in the following directory:
https://github.com/GoogleContainerTools/kpt/tree/main/site/reference/annotations/depends-on

Comment (phanimarupaka) [reply site/book/06-deploying-packages/03-handling-dependencies.md:86]: I remember @etefera fixing the root cause of this. But I am fine with merging this now. @etefera can validate.

**Review by phanimarupaka — APPROVED** [review]

File: site/book/06-deploying-packages/03-handling-dependencies.md (hunk 1 above)
Comment (sdowell) [reply site/book/06-deploying-packages/03-handling-dependencies.md:86]: It appears this issue/discussion may encapsulate what you are describing:
https://github.com/docsifyjs/docsify/issues/1018
